ibis/bigquery/client.py

```diff
@@ -223,12 +223,12 @@ def bq_param_string(param, value):
     return bq.ScalarQueryParameter(param._name, 'STRING', value)
 
 
-@bigquery_param.register(ir.Int64Scalar, six.integer_types)
+@bigquery_param.register(ir.IntegerScalar, six.integer_types)
 def bq_param_integer(param, value):
     return bq.ScalarQueryParameter(param._name, 'INT64', value)
 
 
-@bigquery_param.register(ir.DoubleScalar, float)
+@bigquery_param.register(ir.FloatingScalar, float)
```

Comment:
I've removed those subclasses.

Reply:
Because they are only required to be able to bind methods on them (otherwise even`ScalarExpr(DataType)` and `ColumnExpr(DataType)` would be enough). 

The actual work is done by the datatype passed to `ValueExpr` as an argument. Also see `DataType.scalar` and `DataType.column` class properties.